scanpipe/management/commands/purldb-scan-queue-worker.py

```diff
@@ -0,0 +1,138 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# http://nexb.com and https://github.com/nexB/scancode.io
+# The ScanCode.io software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode.io is provided as-is without warranties.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# Data Generated with ScanCode.io is provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+# ScanCode.io should be considered or used as legal advice. Consult an Attorney
+# for any legal advice.
+#
+# ScanCode.io is a free software code scanning tool from nexB Inc. and others.
+# Visit https://github.com/nexB/scancode.io for support and download.
+
+import time
+import traceback
+
+from django.core.management.base import BaseCommand
+
+from scanpipe.management.commands import AddInputCommandMixin
+from scanpipe.management.commands import CreateProjectCommandMixin
+from scanpipe.pipes import output
+from scanpipe.pipes import purldb
+
+
+class Command(CreateProjectCommandMixin, AddInputCommandMixin, BaseCommand):
+    help = "Create a ScanPipe project."
+
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument(
+            "--sleep",
+            type=int,
+            default=0,
+            action="store",
+            help="Number in seconds how long the loop should sleep for before polling.",
+        )
+
+        parser.add_argument(
+            "--max-loops",
+            dest="max_loops",
+            default=0,
+            action="store",
+            help="Limit the number of loops to a maximum number. "
+            "0 means no limit. Used only for testing.",
+        )
+
+    def handle(self, *args, **options):
```

Comment:
This is too complex and needs to be refactored in smaller pieces

Reply:
I've refactored the code in the `handle()` method. Is what I have still too complex?